packages/bot/src/struct/Command.ts

```diff
@@ -27,7 +38,110 @@ export interface Command<Type extends ApplicationCommandType = ApplicationComman
 	handle: (interaction: InteractionTypeMapping[Type]) => Awaitable<unknown>;
 }
 
-export type CommandConstructor = new (...args: any[]) => Command;
+export interface CommandWithSubcommands {
+	readonly interactionOptions: Omit<CommandBody<ApplicationCommandType.ChatInput>, 'options'>;
+	readonly subcommands: Map<string, Subcommand>;
+	handleAutocomplete?: (interaction: AutocompleteInteraction) => Awaitable<ApplicationCommandOptionChoiceData[]>;
+}
+
+// As of right now, strict typings for the subcommand parameter requires extending SubcommandData
+export interface Subcommand<T extends SubcommandData = SubcommandData>
+	extends Omit<Command<ApplicationCommandType.ChatInput>, 'interactionOptions' | 'handle'> {
+	readonly interactionOptions: Omit<
+		APIApplicationCommandSubcommandOption & { options?: { isMember?: boolean }[] },
+		'type'
+	>;
+	handle: (interaction: ChatInputCommandInteraction<'cached'>, subcommand?: T) => Awaitable<unknown>;
```

Comment:
As described in another review comment,
```suggestion
	handle: (interaction: ChatInputCommandInteraction<'cached'>) => Awaitable<unknown>;
```